From b09f56270cc4cc33835a844b694a499dad9a5fea Mon Sep 17 00:00:00 2001
From: Ryan Zezeski <rpz@joyent.com>
Date: Wed, 19 Jul 2017 17:00:01 -0600
Subject: [PATCH] OS-6227 mac_srs_change_upcall() is unused OS-6240 MAC_DBG_PRT
 is unused Reviewed by: Robert Mustacchi <rm@joyent.com> Reviewed by: Jerry
 Jelinek <jerry.jelinek@joyent.com> Approved by: Jerry Jelinek
 <jerry.jelinek@joyent.com>

---
 usr/src/uts/common/io/mac/mac.c               |  3 ---
 .../uts/common/io/mac/mac_datapath_setup.c    | 26 +------------------
 usr/src/uts/common/sys/mac_impl.h             | 11 --------
 usr/src/uts/common/sys/mac_soft_ring.h        |  2 +-
 4 files changed, 2 insertions(+), 40 deletions(-)

diff --git a/usr/src/uts/common/io/mac/mac.c b/usr/src/uts/common/io/mac/mac.c
index 2895d152a2..1fcc38fb54 100644
--- a/usr/src/uts/common/io/mac/mac.c
+++ b/usr/src/uts/common/io/mac/mac.c
@@ -343,9 +343,6 @@ boolean_t		mac_flow_log_enable;
 boolean_t		mac_link_log_enable;
 timeout_id_t		mac_logging_timer;
 
-/* for debugging, see MAC_DBG_PRT() in mac_impl.h */
-int mac_dbg = 0;
-
 #define	MACTYPE_KMODDIR	"mac"
 #define	MACTYPE_HASHSZ	67
 static mod_hash_t	*i_mactype_hash;
diff --git a/usr/src/uts/common/io/mac/mac_datapath_setup.c b/usr/src/uts/common/io/mac/mac_datapath_setup.c
index e279a6d20a..de7dd0b716 100644
--- a/usr/src/uts/common/io/mac/mac_datapath_setup.c
+++ b/usr/src/uts/common/io/mac/mac_datapath_setup.c
@@ -20,7 +20,7 @@
  */
 /*
  * Copyright (c) 2008, 2010, Oracle and/or its affiliates. All rights reserved.
- * Copyright 2017, Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 #include <sys/types.h>
@@ -1604,30 +1604,6 @@ mac_srs_update_bwlimit(flow_entry_t *flent, mac_resource_props_t *mrp)
 	mac_tx_srs_update_bwlimit(flent->fe_tx_srs, mrp);
 }
 
-void
-mac_srs_change_upcall(void *arg, mac_direct_rx_t rx_func, void *rx_arg1)
-{
-	mac_soft_ring_set_t	*mac_srs = arg;
-	mac_srs_rx_t		*srs_rx = &mac_srs->srs_rx;
-	mac_soft_ring_t		*softring;
-
-	mutex_enter(&mac_srs->srs_lock);
-	ASSERT((mac_srs->srs_type & SRST_TX) == 0);
-	srs_rx->sr_func = rx_func;
-	srs_rx->sr_arg1 = rx_arg1;
-
-	softring = mac_srs->srs_soft_ring_head;
-	while (softring != NULL) {
-		mutex_enter(&softring->s_ring_lock);
-		softring->s_ring_rx_func = rx_func;
-		softring->s_ring_rx_arg1 = rx_arg1;
-		mutex_exit(&softring->s_ring_lock);
-		softring = softring->s_ring_next;
-	}
-
-	mutex_exit(&mac_srs->srs_lock);
-}
-
 /*
  * When the first sub-flow is added to a link, we disable polling on the
  * link and also modify the entry point to mac_rx_srs_subflow_process.
diff --git a/usr/src/uts/common/sys/mac_impl.h b/usr/src/uts/common/sys/mac_impl.h
index 12937cea1b..d98b8e435e 100644
--- a/usr/src/uts/common/sys/mac_impl.h
+++ b/usr/src/uts/common/sys/mac_impl.h
@@ -674,17 +674,6 @@ typedef struct mac_notify_task_arg {
 	mac_ring_t		*mnt_ring;
 } mac_notify_task_arg_t;
 
-/*
- * XXX All MAC_DBG_PRTs must be replaced with call to dtrace probes. For now
- * it may be easier to have these printfs for easier debugging
- */
-#ifdef DEBUG
-extern int mac_dbg;
-#define	MAC_DBG_PRT(a)	if (mac_dbg > 0) {(void) printf a; }
-#else
-#define	MAC_DBG_PRT(a)
-#endif
-
 /*
  * The mac_perim_handle_t is an opaque type that encodes the 'mip' pointer
  * and whether internally a mac_open was done when acquiring the perimeter.
diff --git a/usr/src/uts/common/sys/mac_soft_ring.h b/usr/src/uts/common/sys/mac_soft_ring.h
index 88f1aa7249..e96c41ae91 100644
--- a/usr/src/uts/common/sys/mac_soft_ring.h
+++ b/usr/src/uts/common/sys/mac_soft_ring.h
@@ -22,6 +22,7 @@
 /*
  * Copyright 2010 Sun Microsystems, Inc.  All rights reserved.
  * Use is subject to license terms.
+ * Copyright 2017 Joyent, Inc.
  */
 
 #ifndef	_SYS_MAC_SOFT_RING_H
@@ -652,7 +653,6 @@ extern cpu_t *mac_srs_bind(mac_soft_ring_set_t *, processorid_t);
 extern void mac_rx_srs_retarget_intr(mac_soft_ring_set_t *, processorid_t);
 extern void mac_tx_srs_retarget_intr(mac_soft_ring_set_t *);
 
-extern void mac_srs_change_upcall(void *, mac_direct_rx_t, void *);
 extern void mac_srs_quiesce_initiate(mac_soft_ring_set_t *);
 extern void mac_srs_client_poll_enable(struct mac_client_impl_s *,
     mac_soft_ring_set_t *);
-- 
2.21.0

